lib/DriverTool/swift_parse_test_main.cpp

```diff
@@ -0,0 +1,209 @@
+//===--- swift_parse_test_main.cpp ----------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// A utility tool to measure the parser performance.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/AST/ASTContext.h"
+#include "swift/Basic/LLVM.h"
+#include "swift/Basic/LangOptions.h"
+#include "swift/Bridging/ASTGen.h"
+#include "swift/Parse/Parser.h"
+#include "llvm/Support/CommandLine.h"
+
+#include <chrono>
+#include <ctime>
+#include <numeric>
+
+using namespace swift;
+
+namespace {
+
+struct SwiftParseTestOptions {
+  llvm::cl::opt<bool> SwiftParser =
+      llvm::cl::opt<bool>("swift-parser", llvm::cl::desc("Use SwiftParser"));
+
+  llvm::cl::opt<bool> LibParse =
+      llvm::cl::opt<bool>("lib-parse", llvm::cl::desc("Use libParse"));
+
+  llvm::cl::opt<unsigned int> Iteration = llvm::cl::opt<unsigned int>(
+      "n", llvm::cl::desc("iteration"), llvm::cl::init(1));
+
+  llvm::cl::list<std::string> InputPaths = llvm::cl::list<std::string>(
+      llvm::cl::Positional, llvm::cl::desc("input paths"));
+};
+
+enum class ParseMode {
+  SwiftParser,
+  LibParse,
+};
+
+struct LibParseExecutor {
+  constexpr static StringRef name = "libParse";
+
+  static void performParse(llvm::MemoryBufferRef buffer) {
+    SourceManager SM;
+    unsigned bufferID =
+        SM.addNewSourceBuffer(llvm::MemoryBuffer::getMemBuffer(buffer));
+    DiagnosticEngine diagEngine(SM);
+    LangOptions langOpts;
+    TypeCheckerOptions typeckOpts;
+    SILOptions silOpts;
+    SearchPathOptions searchPathOpts;
+    ClangImporterOptions clangOpts;
+    symbolgraphgen::SymbolGraphOptions symbolOpts;
+    std::unique_ptr<ASTContext> ctx(
+        ASTContext::get(langOpts, typeckOpts, silOpts, searchPathOpts,
+                        clangOpts, symbolOpts, SM, diagEngine));
+
+    SourceFile::ParsingOptions parseOpts;
+    // Always disable body skipping because SwiftParser currently don't have it.
+    // When SwiftParser implements delayed parsing, this should be a command
+    // line option.
+    parseOpts |= SourceFile::ParsingFlags::DisableDelayedBodies;
```

Comment:
Could just add it now, with it being a no-op for the new parser. Probably worth comparing already anyway.

Reply:
Added `-skip-bodies` option.